classes/core/PKPString.php

```diff
@@ -399,23 +402,68 @@ public static function getAmbiguousExtensionsMap()
      * Strip unsafe HTML from the input text. Covers XSS attacks like scripts,
      * onclick(...) attributes, javascript: urls, and special characters.
      *
-     * @param string $input input string
-     * @param string $configKey The config section key['allowed_html', 'allowed_title_html']
+     * @param string|null   $input input string
+     * @param string        $key   The config section key['allowed_html', 'allowed_title_html']
      *
      * @return string
      */
-    public static function stripUnsafeHtml($input, $configKey = 'allowed_html')
+    public static function stripUnsafeHtml(?string $input, string $key = 'allowed_html'): string
     {
-        static $purifier;
-        if (!isset($purifier)) {
-            $config = HTMLPurifier_Config::createDefault();
-            $config->set('Core.Encoding', 'utf-8');
-            $config->set('HTML.Doctype', 'HTML 4.01 Transitional');
-            $config->set('HTML.Allowed', Config::getVar('security', $configKey));
-            $config->set('Cache.SerializerPath', 'cache');
-            $purifier = new HTMLPurifier($config);
+        if (!$input) {
```

Comment:
This is strictly supposed to deal with a `null` case, right? If so, then please use an exact test:
```php
if ($input === null) {
    return '';
}
```
Otherwise, non-string values will get coerced into strings by the function's type hint and the `if` will not behave like it looks at first glance.

Reply:
Done .